.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  golang:
+    env:
+      GO111MODULE: on
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.5'
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-$GITHUB_REPOSITORY-${{ hashFiles('**/go.mod') }}
+      - name: install dependencies
+        run: go mod download
+      - name: go vet
+        run: go vet ./...
+      - name: go test race
+        run: go test -v -race -parallel 4 -coverprofile=covprofile -covermode=atomic ./...
+      - name: install goveralls
```

Comment:
I'm fine with ditching coveralls but that can be discussed and made another PR

Reply:
Agreed, with the mgspack, avro, and some of the generated code there is a lot that won't be tested.